web/src/engine/websites/CoroCoro.ts

```diff
@@ -25,4 +63,69 @@ export default class extends DecoratableMangaScraper {
+            accumulator.push(...mangas);
+            return accumulator;
+        }, []);
+
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL(`${manga.Identifier}`, this.URI)), 'script');
+        const json = this.FindAndParseJSON(scripts, 'omittedMiddleChapters');
+        const { chapters: { earlyChapters, omittedMiddleChapters, latestChapters } } = this.LocateObjectWithNamedProperty<JSONChapters>(json, 'chapters');
+        return [earlyChapters, omittedMiddleChapters, latestChapters].reduce((accumulator: Chapter[], category) => {
+            const chapters = category.map(chapter => new Chapter(this, manga, `/chapter/${chapter.id}`, chapter.title));
+            accumulator.push(...chapters);
+            return accumulator;
+        }, []);
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page<CryptoParams>[]> {
+        const scripts = await FetchCSS<HTMLScriptElement>(new Request(new URL(`${chapter.Identifier}`, this.URI)), 'script');
+        const json = this.FindAndParseJSON(scripts, 'viewerSection');
+        if (!json) {
+            throw new Exception(R.Plugin_Common_Chapter_UnavailableError);
+        }
+        const { viewerSection: { pages } } = this.LocateObjectWithNamedProperty<JSONPages>(json, 'viewerSection');
+        return pages.map(page => new Page<CryptoParams>(this, chapter, new URL(page.src), page.crypto));
+    }
+
+    public override async FetchImage(page: Page<CryptoParams>, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal, true);
+        const cryptoParams = page.Parameters;
+        if (!cryptoParams?.key) return blob;
+        const encrypted = await blob.arrayBuffer();
+        const cipher = { name: 'AES-CBC', iv: Buffer.from(cryptoParams.iv, 'hex') };
+        const cryptoKey = await crypto.subtle.importKey('raw', Buffer.from(cryptoParams.key, 'hex'), cipher, false, ['decrypt']);
+        const decrypted = await crypto.subtle.decrypt(cipher, cryptoKey, encrypted);
+        return Common.GetTypedData(decrypted);
+    }
+
+    private FindAndParseJSON(scripts: HTMLScriptElement[], textToFind: string): JSONObject {
+        const script = scripts.find(script => script.innerText.indexOf(textToFind) > 0)?.innerText;
+        if (!script) return undefined;
+        const json = JSON.parse(script.substring(22, script.length - 2));
+        return JSON.parse(json.substring(json.indexOf(':') + 1));
+    }
+
+    private LocateObjectWithNamedProperty<T>(obj, keyname: string): T {
+        if (!obj) return null;
+        if (obj[keyname]) {
+            return obj as T;
+        }
+        let result = null;
+        for (let i in obj) {
+            if (typeof obj[i] === 'object')
+                result = result ?? this.LocateObjectWithNamedProperty(obj[i], keyname);
```

Comment:
Improve abort condition to short circuit the loop and recursion.
As soon as a result is assigned, then there is no reason to process the loop or any other recursive calls as they won't have any effect, instead return the result
```javascript
result = { x: 0 };
result = result ?? GetStuff(); // return value from GetStuff will never be assigned
```

Reply:
Yeah i thought about breaking from the loop in case result was defined. I deemed it fast enough tho. Okay.